server/src/controllers/post.controllers/create.post.ts

```diff
@@ -29,7 +29,8 @@ export const create = (req: Request, res: Response, next: NextFunction): void =>
             const description = req.body.description;
             const imgPath = req.file.location;
             const hashtagRegex = /\B(\#[a-zA-Z0-9]+\b)/g;
-            const tags = req.body.description.match(hashtagRegex);
+            const hashTags = description ? description.match(hashtagRegex) : [];
+            const tags = hashTags ? hashTags.map((tag: string) => tag.slice(1)) : [];
```

Comment:
description.match(hashtagRegex) is null if it is not match anything.
hashTags is array. Array is always true as array is object.

do you really need [] in ternary expression for hashTags  or maybe 
```const hashTags = description && description.match(hashtagRegex)``
then hashTags has more chances to be null.

Reply:
I've been thinking about how to write this better, thanks. Fixed.